client/src/App.js

```diff
@@ -303,6 +303,24 @@ export default class NormalDisplay extends React.Component {
                 </pre>
     }
 
+    async checkJournalReady() {
+        if (this.isAborted) return;
+        let resp;
+        try {
+            resp = await fetch("/api/check-journal-ready", {
```

Comment:
You can handle this call inside the register call in the server. The client does not need to know be aware of this check.

Reply:
Are you referring to [this call to register()](https://github.com/adobe/asset-compute-devtool/blob/master/server/src/assetComputeDevTool.js#L225) ?

Currently the flow is driven by UI on click of `Run` button. happens only once.

So register() is called by setUpAssetCompute() and immediately check if it is ready. If not(most likely the case) throw error ? 
I think this is only an edge condition hence validation step could suffice. Are you seeing any other benefit out of the refactoring which I am missing?
We have [another ticket](https://jira.corp.adobe.com/browse/NUI-667) to redesign this behavior a bit, so may be can handle it there and remove this altogether then.

Reply:
Yes I was referring to that call to register. I thought the flow would be this:
on page load the UI disables the run button. It does some api requests which end up triggering `setUpAssetCompute()` which calls `register()`. After calling register, we poll the journal until if it is fully set up. (with a timeout of probably a minute or so..). Once that is done, you [return](https://github.com/adobe/asset-compute-devtool/blob/master/server/src/assetComputeDevTool.js#L226) the client (which is now fully set up and ready to run process/ requests). Then the run button is enabled. The benefit of this implementation is it will remove the client having to store the state of the journal and make the code a bit simpler.

I agree with that the [design ticket](https://jira.corp.adobe.com/browse/NUI-667) will replace this implementation so if it is too much effort, I am fine with you leaving the code as is.

I do still have one issue. The behavior now is when `run` is clicked, it will check if the journal is ready by doing one request to the events journal api. if the check fails, it throws an error and does not do a process request. this is not the behavior we want. we want to poll the journal for about 60s (since it can take up to 45s to set up the journal) to keep checking if the journal is ready.

Reply:
I think with the [redesign ticket](https://jira.corp.adobe.com/browse/NUI-667) we will have to  introduce wait 60s in the setup. What you described here could be an alternative. I have linked this comment to jira for reference.